dbms/src/TableFunctions/TableFunctionRandom.cpp

```diff
@@ -0,0 +1,477 @@
+                data[i] = (d / pcg32::max()) * generator();
+            }
+            break;
+        }
+        case TypeIndex::Float64:
+        {
+            auto & data = typeid_cast<ColumnVector<Float64> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            double d = 1.0;
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                d = std::numeric_limits<double>::max();
+                data[i] = (d / pcg32::max()) * generator();
+            }
+            break;
+        }
+        case TypeIndex::Date:
+        {
+            auto & data = typeid_cast<ColumnVector<UInt16> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<UInt16>(generator());
+            }
+            break;
+        }
+        case TypeIndex::DateTime:
+        {
+            auto & data = typeid_cast<ColumnVector<UInt32> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<UInt32>(generator());
+            }
+            break;
+        }
+        case TypeIndex::DateTime64:
+        {
+            UInt32 scale;
+            if (auto * ptype = typeid_cast<const DataTypeDateTime64 *>(type.get()))
+                scale = ptype->getScale();
+            else
+                throw Exception("Static cast to DataTypeDateTime64 failed ", ErrorCodes::BAD_TYPE_OF_FIELD);
+            auto & data = typeid_cast<ColumnDecimal<Decimal64> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                UInt32 fractional = static_cast<UInt32>(generator()) % intExp10(scale);
+                UInt32 whole = static_cast<UInt32>(generator());
+                DateTime64 dt = DecimalUtils::decimalFromComponents<DateTime64>(whole, fractional, scale);
+                data[i] = dt;
+            }
+            break;
+        }
+        case TypeIndex::String:
+        {
+            auto & column_string = typeid_cast<ColumnString &>(column);
+            auto & offsets = column_string.getOffsets();
+            auto & chars = column_string.getChars();
+
+            UInt64 offset = 0;
+            {
+                pcg32 generator(random_seed);
+                offsets.resize(limit);
+                for (UInt64 i = 0; i < limit; ++i)
+                {
+                    offset += 1 + static_cast<UInt64>(generator()) % max_string_length;
+                    offsets[i] = offset - 1;
+                }
+                chars.resize(offset);
+                for (UInt64 i = 0; i < offset; ++i)
+                {
+                    chars[i] = 32 + generator() % 95;
```

Comment:
Missing comments. PS. The method is slightly worse than in @BayoNet 's randomPrintableASCII function.

Reply:
rewritten